security/certora/specs/Governance.spec

```diff
@@ -222,7 +246,10 @@ rule proposal_after_voting_portal_invalidate{
   IGovernanceCore.State state_after = getProposalState(e3, proposalId);
 
   assert !voting_portal_approved_after => (state_before != state_after => isTerminalState(state_after));
-
+  /* ND- what about the other direction
```

Comment:
The 2 properties are equivalent.
P1: !voting_portal_approved_after => (state_before != state_after => isTerminalState(state_after))
P2: (state_before != state_after && !isTerminalState(state_after)) => voting_portal_approved_after
They have the same truth table.

  voting_portal_approved_after | state_before != state_after | isTerminalState(state_after) | P1 | P2 
  ---------------------------------------------------------------------------------------------------
             0                 |              0              |             0                | T  | T 
             0                 |              0              |             1                | T  | T 
             0                 |              1              |             0                | F  | F 
             0                 |              1              |             1                | T  | T 
             1                 |              0              |             0                | T  | T 
             1                 |              0              |             1                | T  | T 
             1                 |              1              |             0                | T  | T 
             1                 |              1              |             1                | T  | T